Srclib.Nuget/ConsoleCommands/DepresolveConsoleCommand.cs

```diff
@@ -10,124 +10,135 @@
-    }
+            await RunResolve(proj.ProjectDirectory);
+            var allDeps = GetAllDeps(proj);
+            return allDeps;
+        }
+
+
+        static IEnumerable<LibraryDescription> GetAllDeps(Project proj) =>
+            proj.GetTargetFrameworks().Select(f => f.FrameworkName)
+            .SelectMany(f =>
+            {
+                var context = new ApplicationHostContext
+                {
+                    Project = proj,
+                    TargetFramework = f
+                };
+                return ApplicationHostContext.GetRuntimeLibraries(context, false).Skip(1); // the first one is the self-reference
+            })
+            .Distinct(LibraryUtils.Comparer)
+            .OrderBy(l => l.Identity?.Name);
+
+
+        static async Task RunResolve(string dir)
+        {
+            var p = new Process();
+            p.StartInfo.WorkingDirectory = dir;
+            p.StartInfo.FileName = _dnuPath.Value;
+            p.StartInfo.Arguments = "restore";
+            p.StartInfo.UseShellExecute = false;
+            p.StartInfo.CreateNoWindow = true;
+            p.StartInfo.RedirectStandardInput = true;
+            p.StartInfo.RedirectStandardOutput = true;
+            p.StartInfo.RedirectStandardError = true;
+            p.Start();
+            // it's important to read stdout and stderr, else it might deadlock
+            var outs = await Task.WhenAll(p.StandardOutput.ReadToEndAsync(), p.StandardError.ReadToEndAsync());
+            p.WaitForExit();
+            // in the future, actually parse output or something
+        }
+
+        static string FindDnuWindows()
+        {
+            return RunForResult(@"C:\Windows\System32\cmd.exe", "/c \"where dnu\"");
+        }
 
-    static string FindDnuWindows()
-    {
-      return RunForResult(@"C:\Windows\System32\cmd.exe", "/c \"where dnu\"");
-    }
+        static string FindDnuNix()
+        {
+            return RunForResult("/bin/bash", "-c \"which dnu\"");
+        }
 
-    static string FindDnuNix()
-    {
-      return RunForResult("/bin/bash", "-c \"which dnu\"");
-    }
+        public static string RunForResult(string shell, string command)
+        {
+            var p = new Process();
+            p.StartInfo.WorkingDirectory = Directory.GetCurrentDirectory();
+            p.StartInfo.FileName = shell;
+            p.StartInfo.Arguments = command;
+            p.StartInfo.UseShellExecute = false;
+            p.StartInfo.CreateNoWindow = true;
+            p.StartInfo.RedirectStandardInput = true;
+            p.StartInfo.RedirectStandardOutput = true;
+            p.StartInfo.RedirectStandardError = false;
+
+            p.Start();
+            var result = p.StandardOutput.ReadToEnd().Trim();
+            p.WaitForExit();
+            return result;
+        }
+
+        public static string GetDll(string path)
+        {
+            return RunForResult("/bin/bash", "-c \"cd " + path + " && find -name *.dll | head -1\"");
```

Comment:
Yes, it is hacky indeed. For some reason, dnx interface doesn't resolve a dependency to an actual dll file. Instead it resolves a dependency to a some folder, that contains multiple same-named dlls for different platforms (and sometimes different versions), e. g.

~/.dnx/packages/System.Reflection/4.0.10# find -name "*.dll"

./lib/DNXCore50/System.Reflection.dll
./lib/netcore50/System.Reflection.dll
./ref/dotnet/System.Reflection.dll
./runtimes/win8-aot/lib/netcore50/System.Reflection.dll

This is really inconvenient, but I didn't find any straightforward ways to deal with it in dnx interface. I'll try to dig in roslyn sources to see how it solves this problem (obviously roslyn needs to solve exactly the same problem). But for now i have to do it through bash invocation.